src/adapter/adapter.ts

```diff
@@ -110,7 +110,18 @@ abstract class Adapter extends events.EventEmitter {
                             serialPortOptions.path = `tcp://${mdnsIp}:${mdnsPort}`;
                             serialPortOptions.adapter = mdnsAdapter;
                             serialPortOptions.baudRate = mdnsBaud;
-                            resolve(new adapter(networkOptions, serialPortOptions, backupPath, adapterOptions, logger));
+                            
+                            if (adapterLookup.hasOwnProperty(serialPortOptions.adapter) 
+                            && serialPortOptions.adapter !== 'auto') {
```

Comment:
`adapterLookup` doesn't contain `auto` so the `serialPortOptions.adapter !== 'auto'` statement is not needed

Reply:
I understand that, but typescript doesn't
![image](https://github.com/Koenkk/zigbee-herdsman/assets/31830530/a130b287-87ad-410e-b3bc-371295a40e7e)
![image](https://github.com/Koenkk/zigbee-herdsman/assets/31830530/d39fb6c3-a245-4a03-900d-36965f248a3a)